src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/ResultItem.tsx

```diff
@@ -303,24 +309,35 @@ export const ResultItem: React.FC<ResultItemProps & AccessibilityProps> = ({
             <DestinationIcon style={styles.iconContainer} />
             <View style={styles.departureTimes}>
               <ThemeText type="body__tertiary" color="primary" testID="endTime">
-                {formatToClock(tripPattern.expectedEndTime, language, 'ceil')}
+                {(legs.length && isLegFlexibleTransport(legs[legs.length - 1])
+                  ? t(TripDetailsTexts.trip.leg.circaLabel)
+                  : '') +
+                  formatToClock(tripPattern.expectedEndTime, language, 'ceil')}
               </ThemeText>
             </View>
           </View>
         </View>
-        <ResultItemFooter bookingRequirement={tripPattern.bookingRequirement} />
+        <ResultItemFooter tripPattern={tripPattern} />
       </Animated.View>
     </TouchableOpacity>
   );
 };
 
 const ResultItemFooter: React.FC<{
-  bookingRequirement: BookingRequirement;
-}> = ({bookingRequirement}) => {
+  tripPattern: TripPattern;
+}> = ({tripPattern}) => {
   const styles = useThemeStyles();
   const {t} = useTranslation();
 
-  const {requiresBooking, requiresBookingUrgently} = bookingRequirement;
+  const now = useNow(2500);
+
+  const requiresBookingUrgently = getTripPatternRequiresBookingUrgently(
```

Comment:
Seems like this function is only used here, consider move it to an internal function, instead of in `utils/`.